torch/_inductor/scheduler.py

```diff
@@ -4278,6 +4279,99 @@ def get_graph_partition_signature(
 
         return signatures[::-1]
 
+    def reorder_for_minimizing_partition(
+        self,
+        nodes: list[BaseSchedulerNode],
+    ) -> list[BaseSchedulerNode]:
+        """
+        Reorder nodes to minimize the number of partitions via a bfs
+        topological sort. This is the optimal reodering such that the
```

Comment:
This is the optimal number of partitions, but it's not optimal with respect to the other things we might want to optimize for..